Indicators/TrueRange.cs

```diff
@@ -63,10 +63,10 @@ public TrueRange(string name)
         /// <returns>A new value for this indicator</returns>
         protected override decimal ComputeNextValue(IBaseDataBar input)
         {
-            if (!IsReady)
+            if (Samples == 1)
             {
                 _previousInput = input;
-                return 0m;
+                return input.High - input.Low;
```

Comment:
Returning `input.High - input.Low` seems okay when `samples == 1`, it does not mean the indicator is ready and warmed up though, it requires 1 extra data point given it's comparing current with previous, don't agree with the changes at WarmUpPeriod/IsReady

Think `AverageTrueRange` above should probably look whether TrueRange is ready on IsReady too

Reply:
As exposed https://github.com/QuantConnect/Lean/pull/8149#discussion_r1670991914 my opinion is that "Ready" or not is not enough to describe an indicator state... this PR was a attempt to show case how a refactor could be done but you are right for not accepting in its current state.

I think at least 3 states are required.

 - Cold
 - WarmingUp
 - Ready

Cold could be indicator default state when no data have been received
WarmingUp when 1 data have been received (in the case of TrueRange)
Ready when 2 or more data have been received

Any opinion?

Reply:
Hey! Agree that potentially there could be other states (even failing/invalid), but if we did that, it should be handled generically for all indicators, don't think it's the scope of these changes. Let's follow current pattern, IsReady means it's fully ready, which is expected to be true when received the WarmupPeriod amount of data count, not before

Reply:
> Agree that potentially there could be other states (even failing/invalid), but if we did that, it should be handled generically for all indicators, don't think it's the scope of these changes/pr.

I see u added state for the AvgTR. This should be solve generically in it's own PR, not related to this one, after a design/brain storm I think